data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/codec/CSVCodec.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source.codec;
+
+import com.amazon.dataprepper.model.annotations.DataPrepperPlugin;
+import com.amazon.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.log.JacksonLog;
+import com.amazon.dataprepper.model.record.Record;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.MappingIterator;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvReadException;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+@DataPrepperPlugin(name = "csv", pluginType = Codec.class, pluginConfigurationType = CSVCodecConfig.class)
+public class CSVCodec implements Codec {
```

Comment:
Let's use `Csv` instead of `CSV`. I think this is more consistent with the current code base and Java conventions. Please change other classes as well.

Reply:
I agree that this change is more consistent with conventions and the code base. I suppose we should also change the `CSVProcessor` class to `CsvProcessor` — is it okay to include these changes in this pull request? I can also open a new pull request to change to `CsvProcessor` if it's out of scope of this pull request.

Reply:
A different PR would be better. That will help get this one in faster.